### PR TITLE
SUP-2343: Add lists of Attributes that can be used with `buildkite-agent step` 

### PR DESCRIPTION
**Pull request by tomowatt**
Had a customer reach out checking which attributes could be used with the `buildkite-agent step [get|update]` commands as they found it unclear from the docs. So adding list of available Attributes for clarity.
Will need to also update the Agents docs - https://github.com/buildkite/agent/blob/d387c3bdd58eb3fd94906e3e400ca8e0ab230e87/clicommand/step_get.go#L26-L31 - to correct example as `retry` is not an available attribute at the time of writing.

### CODE REVIEW

**Comment by tomowatt** [comment]
Will update PR, if code changes in Agent are merged: https://github.com/buildkite/agent/pull/2879

**Comment by tomowatt** [comment]
Agent v3.75.0 released with changes to help output

File: pages/agent/v3/cli_step.md [hunk 1]
```diff
@@ -4,13 +4,31 @@ The Buildkite agent's `step` command provides the ability to retrieve and update
 
 ## Updating a step
 
-Use this command in your build scripts to update an attribute of a step.
+Use this command in your build scripts to update an attribute of a step. The following attributes can be updated:
+
+* `label`
```
Comment (gilesgas) [inline pages/agent/v3/cli_step.md:9]: A list containing only a single item looks rather odd. Unless we're intending to add more items to this list in the very near future, I'd recommend simplifying this to:

```suggestion
Use this command in your build scripts to update the `label` attribute of a step.
```

**Review by gilesgas — APPROVED** [review]
Thanks very much for preparing this @tomowatt !

Happy for you to merge this in once you've committed my suggestion (unless of course more attributes are to be added to the list for `buildkite-agent step update ...`).

I'll fix up the wording for the agent description (i.e. in the agent repo) in a separate PR.

**Pull request merged**